src/acl/FilledChecklist.cc

```diff
@@ -218,9 +218,15 @@ ACLFilledChecklist::ACLFilledChecklist(const acl_access *A, HttpRequest *http_re
     rfc931[0] = '\0';
 
     changeAcl(A);
+    setRequest(http_request);
+    setIdent(ident);
+}
 
-    if (http_request != NULL) {
-        request = http_request;
+void ACLFilledChecklist::setRequest(HttpRequest *httpRequest)
+{
+    assert(!request);
```

Comment:
This assert does not look right. It is ensuring that the setRequest() can only overwrite previous request values, and never sets the first one.

Reply:
@yadij, please take a second look. I believe the code (intensionally) does the opposite of what you say it does -- it checks that setRequest() is not called for an already configured checklist. Besides, this code worked in lab tests, and it could not have worked if your interpretation was correct AFAICT.

Reply:
doh, you are right. please ignore this one